controller/linodemachine_controller_helpers.go

```diff
@@ -345,6 +376,84 @@ func getFirewallID(ctx context.Context, machineScope *scope.MachineScope, logger
 	return *linodeFirewall.Spec.FirewallID, nil
 }
 
+func getNextIP(ips []string, prefixStr string) string {
+	prefix := netip.MustParsePrefix(prefixStr)
+	currentIp := prefix.Addr().Next()
+
+	ipString := currentIp.String()
+	for {
+		if !slices.Contains(ips, ipString) {
+			break
+		}
+		currentIp = currentIp.Next()
+		ipString = currentIp.String()
+	}
+	return ipString
+}
+
+func reserveNextIP(ctx context.Context, machineScope *scope.MachineScope, logger logr.Logger) (string, error) {
+	namespace := machineScope.Cluster.Namespace
+	clusterName := machineScope.Cluster.Name
+	var nextIP string
+	var ipsMap corev1.ConfigMap
```

Comment:
You'll need to update linodemachine_controller.go to include configmaps for the kubebuilder RBAC annotation

Reply:
done, thanks!